cmd/serverless-init/cloudservice/cloudrun.go

```diff
@@ -37,31 +43,44 @@ type CloudRun struct {
 // GetTags returns a map of gcp-related tags.
 func (c *CloudRun) GetTags() map[string]string {
 	tags := metadataHelperFunc(helper.GetDefaultConfig()).TagMap(c.spanNamespace)
+	tags["origin"] = c.GetOrigin()
+	tags["_dd.origin"] = c.GetOrigin()
+
 	revisionName := os.Getenv(revisionNameEnvVar)
 	serviceName := os.Getenv(ServiceNameEnvVar)
 	configName := os.Getenv(configurationNameEnvVar)
-
 	if revisionName != "" {
-		tags[c.spanNamespace+"revision_name"] = revisionName
 		tags["revision_name"] = revisionName
+		if c.spanNamespace == cloudRunService {
+			tags[runRevisionName] = revisionName
+		} else {
+			tags[functionRevisionName] = revisionName
+		}
 	}
 
 	if serviceName != "" {
-		tags[c.spanNamespace+"service_name"] = serviceName
 		tags["service_name"] = serviceName
+		if c.spanNamespace == cloudRunService {
+			tags[runServiceName] = serviceName
+		} else {
+			tags[functionServiceName] = serviceName
+		}
 	}
 
 	if configName != "" {
-		tags[c.spanNamespace+"configuration_name"] = configName
 		tags["configuration_name"] = configName
+		if c.spanNamespace == cloudRunService {
+			tags[runConfigName] = configName
+		} else {
+			tags[functionConfigName] = configName
+		}
 	}
 
 	if c.spanNamespace == cloudRunFunction {
-		tags = c.getFunctionTags(tags)
+		return c.getFunctionTags(tags)
 	}
 
-	tags["origin"] = c.GetOrigin()
-	tags["_dd.origin"] = c.GetOrigin()
+	tags["_dd.gcr.resource_name"] = "projects/" + tags["project_id"] + "/locations/" + tags["location"] + "/services/" + serviceName
```

Comment:
Just confirming that this is intended to always be added, even for a gcrfx function?  It looks like functions will end up with `_dd.gcr.resource_name` and `_dd.gcrfx.resource_name`. I don't think this is a backwards compatible addition so I would think we only need one. Can you help me understand why both are here?